docs/antora/modules/ROOT/pages/Library_Dependencies.adoc

```diff
@@ -150,13 +150,61 @@ def compileIvyDeps = Agg(
 )
 ----
 
-It is also possible to use higher version of the same library dependencies already  defined in `ivyDeps`, to ensure you compile against a minimal API version, but actually run with the latest available version.
+It is also possible to use higher version of the same library dependencies already defined in `ivyDeps`, to ensure you compile against a minimal API version, but actually run with the latest available version.
+
+== Detecting transitive dependencies
+
+To render a tree of dependencies (transitive included) and possible upgrades for a given module, you can run `mill myModule.ivyDepsTree`.
+
+You can also compile your module(s) and examine files such as `ivyDeps.json` and `transitiveIvyDeps.json` in your `out` build's folder for a given module. `ivyDepsTree.log` contains file output of the above ivyDepsTree command too.
+
+You can observe the actual version being used by running `mill show myModule.resolvedIvyDeps`. If you run `mill myModule.resolvedIvyDeps`, the same information is available in `out/myModule/resolvedIvyDeps.json`.
 
 == Excluding transitive dependencies
 
 You can use the `.exclude` method on a dependency. It accepts  `organization` and `name` tuples, to be excluded.
 Use the special name `*` to match all ``organization``s or ``name``s.
 
+.Example: Exclude `fansi_2.12` library from transitive dependency set of `pprint`.
+[source,scala]
+----
+def deps = Agg(
+  ivy"com.lihaoyi::pprint:0.5.3".exclude("com.lihaoyi" -> "fansi_2.12")
+)
+----
+
+You can also use `.excludeOrg` or `excludeName`:
+
+
+.Example: Exclude all `com.lihaoyi` libraries from transitive dependency set of `pprint`.
+[source,scala]
+----
+val deps = Agg(ivy"com.lihaoyi::pprint:0.5.3".excludeOrg("com.lihaoyi"))
+----
+
+Note: You can chain multiple exclusions with `exclude`, `excludeOrg`, and `excludeName`.
+
+.Example: Excluding a library (fansi) by name from transitive dependency set of `pprint`.
+[source,scala]
+----
+val deps = Agg(
+  ivy"com.lihaoyi::pprint:0.5.3"
+    .excludeName("fansi_2.12")
+    .excludeName("sourcecode")
+)
+----
+
+== Forcing versions
+
+You can use the `forceVersion` method to ensure the used version of a dependency is what you have declared.
+
+- You declare a dependency `val deps = Agg(ivy"com.lihaoyi::fansi:0.2.14")`
+- There is another dependency, `val deps = Agg(ivy"com.lihaoyi::PPrint:0.8.1")`
+- PPrint 0.8.1 uses fansi 0.4.0, so it is a transitive dependency
+- `mill show myModule.resolvedIvyDeps | grep "fansi"` should show fansi 0.4.0
+- If you want to force to the older version (to prevent it being evicted, and replaced by 0.4.0) then you can use `val deps = Agg(ivy"com.lihaoyi::fansi:0.2.14".forceVersion())`
+- `mill show myModule.resolvedIvyDeps | grep "fansi"` should show fansi 0.2.14
```

Comment:
I'm not sure if it's a great practice (default behaviour is to evict old versions in favour of newer transitive versions, it seems) but the option is there.

Maybe not for this PR, but it seems there are no docs with the word "evict" in. It would be handy to know more about what happens here, and what best practices there are to get the most stable versions.

Tangent:
I believe Scala Steward only shows specific dependency updates (not transitive changes).
Given that mill evicts old versions in favour of newer transitive ones, Steward PRs can actually bring in more changes than you expect.

Reply:
You're right, `forceVersion` is not really recommended. It help get things done, but it is problematic, as it is not properly propagated to published artifacts. We have an open issue for that: https://github.com/com-lihaoyi/mill/issues/1975

It's best to treat it as experiemental.

Reply:
Can you please use `*` for the list items?

Reply:
Thanks, I thought some linter for adoc might catch the bullet point preffered type. IntelliJ's adoc plugin renders both fine.

I added a caution tooltip for forcing versions. It could help people toy with versions locally, so seems worth including with the above caveat mentioned.

Reply:
Great.

The `-` for lists is perfect legal Asciidoc syntax, but we use the `*` syntax in all the other places and it works better for nested lists.